certbot/certbot/compat/filesystem.py

```diff
@@ -232,39 +270,38 @@ def makedirs(file_path, mode=0o777):
     """
     Rewrite of original os.makedirs function, that will ensure on Windows that given mode
     is correctly applied.
+
     :param str file_path: The file path to open
     :param int mode: POSIX mode to apply on leaf directory when created, Python defaults
                      will be applied if ``None``
     """
-    if POSIX_MODE:
-        # Since Python 3.7, os.makedirs does not set the given mode to the intermediate directories
-        # that could be created in the process. To keep things safe and consistent on all
-        # Python versions, we set the umask accordingly to have all directories (intermediate and
-        # leaf) created with the given mode.
-        current_umask = os.umask(0)
+    current_umask = umask(0)
+    try:
+        if POSIX_MODE:
+            # Since Python 3.7, os.makedirs does not set the given mode to the intermediate
+            # directories that could be created in the process. To keep things safe and consistent
+            # on all Python versions, we set the umask accordingly to have all directories
+            # (intermediate and leaf) created with the given mode.
+            umask(current_umask | 0o777 ^ mode)
```

Comment:
I think we need to do something similar to this in the Windows case.

Reply:
It was the intention indeed. Done.